p8e-api-webservice/src/main/kotlin/io/provenance/p8e/webservice/config/AppConfig.kt

```diff
@@ -78,4 +89,42 @@ class AppConfig : WebMvcConfigurer {
             )
         }
     }
+
+    /**
+     * Add support for new key management.
+     */
+    @Bean
+    @Scope(value = ConfigurableBeanFactory.SCOPE_SINGLETON)
+    fun smartKeyApiClient(smartKeyProperties: SmartKeyProperties): ApiClient? {
+        return if(smartKeyProperties.apiKey != null && smartKeyProperties.apiKey != "") {
+            ApiClient().apply {
+                setBasicAuthString(smartKeyProperties.apiKey)
+                com.fortanix.sdkms.v1.Configuration.setDefaultApiClient(this)
+
+                // authenticate with api
+                val authResponse = authenticationApi(this).authorize()
+                val auth = this.getAuthentication("bearerToken") as ApiKeyAuth
+                auth.apiKey = authResponse.accessToken
+                auth.apiKeyPrefix = "Bearer"
+            }
+        } else {
+            null // SmartKey is not initialized.
+        }
+    }
+
+    @Bean
+    fun authenticationApi(smartKeyApiClient: ApiClient): AuthenticationApi = AuthenticationApi(smartKeyApiClient)
```

Comment:
I guess these beans below don't complain when smartKeyApiClient is null?

Reply:
im going to clean this up, and just pass an uninstantiated instance of the API so that at least we know what the problem is if there are issues.